apps/web/src/components/nav/RootNavMenu.tsx

```diff
@@ -44,7 +46,7 @@ export const RootNavMenu: React.FC = () => {
   return (
     <NavMenu variant="root">
       <NavMenuSection>
-        <OrganizationSelect />
+        {IS_CLERK_AUTH_ENABLED ? <OrganizationSwitcher /> : <OrganizationSelect />}
```

Comment:
How can we remove this ternary duplication in the codebase? Should we use a facade component such ash <OrganizationSwitcher/> that does the ternary inside, or maybe a react higher-order component (i.e. `withAuthComponent(OrgSwitcher, OrgSelect)`  to do the feature flag check in there and return argument 1 or 2?

Reply:
Both options are feasible, I think the questions is just how nested will the if/else be. For now I tried to follow existing conventions and keep it as simple as possible. 

What could also matter is how different the EE version will be - e.g. if we plan to change the `RootNavMenu` more, it could make more sense to have `EERootNavMenu` cause we are hiding the settings nav item here.